File: Javascript/0876_Middle_of_the_Linked_List.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var middleNode = function(head) {
    let count = 1;
    let mid = head;
    while(head.next) {
        head = head.next;
        count++;
        if(count % 2 == 0) {
            mid = mid.next;
        }
    }
    return mid;
};

File: Python/0876_Middle_of_the_Linked_List.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def middleNode(self, head):
        """
        :type head: ListNode
        :rtype: ListNode
        """
        p = head
        count = 0
        while head.next:
            head = head.next
            if count%2 == 0:
                p = p.next
            count += 1
        return p
